### PR TITLE
mulrIb

### PR DESCRIPTION
**Pull request by CohenCyril**
##### Motivation for this change

Adding `mulrIb` a specific injectivity case of `*+` when the lhs is a boolean.

##### Things done/to do

- [x] added corresponding entries in `CHANGELOG_UNRELEASED.md` (do not edit former entries)
- ~added corresponding documentation in the headers~

##### Automatic note to reviewers

Read [this Checklist](https://github.com/math-comp/math-comp/wiki/Checklist-for-following,-reviewing-and-playing-with-a-PR#checklist-for-reviewing-a-pr) and make sure there is a milestone.

### CODE REVIEW

**Comment by chdoc** [comment]
This looks a bit bespoke to me, what's the motivation or where is this useful?

**Comment by CohenCyril** [comment]
https://github.com/CohenCyril/signdet/blob/master/signdet.v#L318, ideally that would be `mulrn` is injective, but that's true only if the characteristic of the ring is `0`.

**Review by chdoc — APPROVED** [review]
LGTM

**Comment by chdoc** [comment]
@CohenCyril I can't merge this, so what should I do now? Who, apart from you, is most familiar with that part of the library?

**Comment by CohenCyril** [comment]
woops, I think @thery, @gares, @pi8027, @strub might all be good candidates

**Comment by thery** [comment]
@CohenCyril sorry too cryptic for me, don't see how this could be a reusable theorem.

**Comment by CohenCyril** [comment]
> @CohenCyril sorry too cryptic for me, don't see how this could be a reusable theorem.

Well I do think it is reusable. Let me try to explain it another way.

The construction `x *+ b` is `x` if `b` otherwise `0`, it is frequently used in combination with membership (e.g. `x \in G`) or Kronecker symbol (`i == j`), the latter gives a "Dirac" function `fun j => x *+ (i == j)` which is thus injective through `mulrIb`.
This is used pervasively in the representation of diagonal matrices `\matrix_(i, j) (d 0 i *+ (i == j))`, and this lemma helps identifying nonzero values of the matrix with elements of the diagonal.

In other words again, this lemma, explains these "Dirac/weight function" are injective on  `b` as long as `x` is nonzero, which allows to infer the validity of `b` (i.e.  `i == j`, or `i \in G`, depending on the use cases) from the nullity of `x`.

**Comment by thery** [comment]
@CohenCyril I've understood what it is your use case, but I just doubt it will be reused in other places.

**Comment by CohenCyril** [comment]
What is the conclusion?

**Comment by thery** [comment]
not a fan

File: mathcomp/algebra/ssralg.v [hunk 1]
```diff
@@ -835,6 +835,9 @@ Qed.
 Lemma mulrnAC x m n : x *+ m *+ n = x *+ n *+ m.
 Proof. by rewrite -!mulrnA mulnC. Qed.
 
+Lemma mulrIb (x : V) : x != 0 -> injective (fun b : bool => x *+ b).
```
Comment (pi8027) [inline mathcomp/algebra/ssralg.v:838]: I have no opinion on how it is useful, but this should be exported from `GRing.Theory`.

**Comment by chdoc** [comment]
@thery @CohenCyril As the assignee for this PR, I would be happy if we could make progress towards either merging or closing this PR. Personally, I don't think this one-liner will hurt, so I would be in favor of merging it, even if it may be slightly bespoke. 

@CohenCyril How often is this lemma used in your developments?

**Comment by thery** [comment]
@chdoc as I said I am not a big fan, but it will not keep me from sleeping at night if it is merged. My only point if If it is a general construction it should at least be documented. 
As a naive user I would  have missed the theorem, as I would have coded the gard as
`(b%:R * e)` as I feel more at ease manipulating `*` than `*+`
